src/Hipace.cpp

```diff
@@ -727,10 +726,14 @@ Hipace::PredictorCorrectorSolveOneSubSlice (const int lev, const int step, const
 }
 
 void
-Hipace::ResetAllQuantities ()
+Hipace::ResetAllQuantities (const int step)
 {
     HIPACE_PROFILE("Hipace::ResetAllQuantities()");
 
+    if (step != m_numprocs_z - 1 - m_rank_z && m_multi_laser.m_use_laser) {
+        ResetLaser();
+    }
+
```

Comment:
Why not always resetting the laser slices when starting the head box?

Furthermore, you can use `m_use_laser` (from `Hipace` class) instead of `m_multi_laser.m_use_laser` I believe. If so, you don't need to pass `step` as a function argument.

```suggestion
    if (m_use_laser) ResetLaser();

```